src/esp/sim/Simulator.cpp

```diff
@@ -117,171 +126,405 @@ void Simulator::reconfigure(const SimulatorConfiguration& cfg) {
+
+  // TODO : reset may eventually have all the scene instance instantiation code
+  // so that scenes can be reset
+  reset();
+  return true;
+}  // Simulator::createSceneInstanceNoRenderer
+
+bool Simulator::createPathfinder(const std::string& navmeshFilename) {
+  // create pathfinder and load navmesh if available
+  pathfinder_ = nav::PathFinder::create();
+  if (Cr::Utility::Directory::exists(navmeshFilename)) {
+    LOG(INFO) << "Simulator::createPathfinder : Loading navmesh from "
+              << navmeshFilename;
+    pathfinder_->loadNavMesh(navmeshFilename);
+    LOG(INFO) << "Simulator::createPathfinder : Loaded.";
+    return true;
+  }
+  LOG(WARNING) << "Simulator::createPathfinder : Navmesh file not found, "
+                  "checked at filename : '"
+               << navmeshFilename << "'";
+  return false;
+}  // Simulator::createPathfinder
+
+bool Simulator::loadSemanticSceneDescriptor(
+    const std::string& semanticSceneDescFilename,
+    const assets::AssetType& assetType) {
+  namespace FileUtil = Cr::Utility::Directory;
 
+  // semantic scene descriptor might not exist, so
   semanticScene_ = nullptr;
   semanticScene_ = scene::SemanticScene::create();
-  switch (stageType) {
-    case assets::AssetType::INSTANCE_MESH:
-      houseFilename = Cr::Utility::Directory::join(
-          Cr::Utility::Directory::path(houseFilename), "info_semantic.json");
-      if (io::exists(houseFilename)) {
-        scene::SemanticScene::loadReplicaHouse(houseFilename, *semanticScene_);
+  bool present = false;
+  switch (assetType) {
+    case assets::AssetType::INSTANCE_MESH: {
+      const std::string tmpFName = FileUtil::join(
+          FileUtil::path(semanticSceneDescFilename), "info_semantic.json");
+      if (FileUtil::exists(tmpFName)) {
+        present =
+            scene::SemanticScene::loadReplicaHouse(tmpFName, *semanticScene_);
+      } else {
+        // Provide warning msgs for failures?
+        present = false;
       }
       break;
-    case assets::AssetType::MP3D_MESH:
+    }
+    case assets::AssetType::MP3D_MESH: {
       // TODO(msb) Fix AssetType determination logic.
-      if (io::exists(houseFilename)) {
+      if (FileUtil::exists(semanticSceneDescFilename)) {
         using Corrade::Utility::String::endsWith;
-        if (endsWith(houseFilename, ".house")) {
-          scene::SemanticScene::loadMp3dHouse(houseFilename, *semanticScene_);
-        } else if (endsWith(houseFilename, ".scn")) {
-          scene::SemanticScene::loadGibsonHouse(houseFilename, *semanticScene_);
+        if (endsWith(semanticSceneDescFilename, ".house")) {
+          present = scene::SemanticScene::loadMp3dHouse(
+              semanticSceneDescFilename, *semanticScene_);
+        } else if (endsWith(semanticSceneDescFilename, ".scn")) {
+          present = scene::SemanticScene::loadGibsonHouse(
+              semanticSceneDescFilename, *semanticScene_);
+        } else {
+          present = false;
```

Comment:
Since present defaults to false, are these else ```present = false``` even needed? All these switches are mutually exclusive. Feels like it could mask an issue.

Reply:
Here again I parsed the code into an individual method in the interest of future development, when it does not really need to be split from where it is called right now, so I'll return this code to where it was originally executed.